### PR TITLE
PD-305: Port <Toggle> to TypeScript

### PR DESCRIPTION
**Pull request by jetpacmonkey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [PD-305: Port <Toggle> to TypeScript](https://jira.mongodb.org/browse/PD-305)

## 🛠 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/Checkbox/package.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
   "description": "LeafyGreen UI Kit Checkbox",
   "main": "./dist/index.js",
   "scripts": {
-    "build": "../../node_modules/.bin/webpack --config ../../webpack.config.js"
+    "build": "../../node_modules/.bin/webpack --config ../../webpack.config.js",
+    "ts:emit": "tsc --project ./tsconfig.json"
```
Comment (jetpacmonkey) [inline packages/Checkbox/package.json]: Drive-by fix adding a line that I missed in #82

Comment (jetpacmonkey) [reply packages/Checkbox/package.json]: Also... Is there some way we could lint for this?

File: packages/Toggle/src/Toggle.tsx [hunk 2]
```diff
@@ -40,19 +51,26 @@ const focusStateStyle = css`
   }
 `;
 
-const getContainerStyles = ({ size, disabled }) => {
-  const sizeStyle = {
-    default: css`
+interface StateForStyle {
```
Comment (jetpacmonkey) [inline packages/Toggle/src/Toggle.tsx]: I don't love the name `StateForStyle`. Any suggestions on a better one?

File: packages/Checkbox/package.json (hunk 1 above)
Comment (hswolff) [reply packages/Checkbox/package.json]: Would be neat to make an eslint rule for this. Or just a custom script to validate that we have this field present in a package.json file. But yeah agree, annoying we have to manually catch this.

File: packages/Toggle/src/Toggle.tsx (hunk 2 above)
Comment (hswolff) [reply packages/Toggle/src/Toggle.tsx]: Meh, it works for me. Can't think of anything better either.

File: packages/Checkbox/package.json (hunk 1 above)
Comment (DesignerDave) [reply packages/Checkbox/package.json]: Even better – re-explore removing these redundant scripts. They're used to work around lerna, but it would be great to revisit this to remove the `build`/`ts:emit` scripts, as well as the package-specific `tsconfig.json`.

File: packages/Toggle/src/Toggle.tsx (hunk 2 above)
Comment (DesignerDave) [reply packages/Toggle/src/Toggle.tsx]: Possibly `StatefulStyleProps`? Not sure that's better?

File: packages/Toggle/src/Toggle.tsx [hunk 3]
```diff
@@ -403,12 +444,13 @@ export default class Toggle extends PureComponent {
     }
   };
   render() {
-    const checkboxId = this.props.id || this.checkboxId;
+    const checkboxId = this.props.id
+      ? this.props.id.toString()
```
Comment (DesignerDave) [inline packages/Toggle/src/Toggle.tsx]: Is the ID here not converted to a string when rendered in HTML? Could be wrong but I don't think this is necessary.

Comment (jetpacmonkey) [reply packages/Toggle/src/Toggle.tsx]: I think you're right, but TS complained, so 🤷‍♂

Comment (DesignerDave) [reply packages/Toggle/src/Toggle.tsx]: I'm not in love for altering the output JS for the sake of TS complaining. It would be good to know specifically why it thinks that it can't be a number.
